docs/articles/documentation/extending-testcafe/custom-reporter/reporter-methods.md

```diff
@@ -0,0 +1,194 @@
+---
+layout: docs
+title: Reporter Methods
+permalink: /documentation/extending-testcafe/custom-reporter-plugin/reporter-methods.html
+checked: false
+---
+# Reporter Methods
+
+To create your [custom reporter](index.md#implementing-the-reporter), you need to implement the following reporter methods.
+
+> You can use [helper methods](helper-methods.md) within the reporter methods to output required data.
+
+## reportTaskStart
+
+Fires when a test task starts, which happens at the beginning of testing.
+
+```text
+reportTaskStart (startTime, userAgents, testCount)
+```
+
+Parameter    | Type             | Description
+------------ | ---------------- | ---------------------------------------
+`startTime`  | Date             | The date and time when testing started.
+`userAgents` | Array of Strings | The list of browsers used for testing.
+`testCount`  | Number           | The total number of tests to run.
+
+**Example**
+
+```js
+reportTaskStart (startTime, userAgents, testCount) {
+    this.startTime = startTime;
+    this.testCount = testCount;
+
+    function formatDate (d) {
```

Comment:
Oh, my... You've invested so much labor in it... Cool...

I really hate to say it - but why not `startTime.toLocaleString()` or any of these

[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#Conversion_getter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#Conversion_getter)